sys/net/application_layer/asymcute/asymcute.c

```diff
@@ -879,8 +879,8 @@ int asymcute_publish(asymcute_con_t *con, asymcute_req_t *req,
         goto end;
     }
 
-    /* get message id */
-    req->msg_id = _msg_id_next(con);
+    /* get message id, set to 0 for QOS 0 */
+    req->msg_id = (flags & MQTTSN_QOS_MASK) ? _msg_id_next(con) : 0;
```

Comment:
From OP:

> The MQTT-SN standard states in 5.4.12 for PUBLISH messages: "MsgId: same meaning as the MQTT “Message ID”; only relevant in case of QoS levels 1 and 2, otherwise coded 0x0000.". 

So it must also be 0 for QoS -1 (_negative 1_), right? So there also needs to be a check if both QoS flags are set.

Reply:
Yes. did forget about QOS -1. Although it is not really applicable for the current implementation, as only QOS1 is supported and hence `MQTTSN_QOS_0` and `MQTTSN_QOS_1` are the only two flags currently allowed in that part of the flags field... But does not hurt to be more specific here in any case, will fix.